### PR TITLE
Style checkboxes in supporting browsers.

### PR DESCRIPTION
**Pull request by huwshimi**

### CODE REVIEW

**Comment by huwshimi** [comment]
QA:
- in Chrome view the gui with `/:flags:/react/`
- add a service
- navigate to the list of units
- click on a checkbox
- check that they have custom styles (should turn green when clicked)

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/3161/
Test PASSed.

File: jujugui/static/gui/src/app/assets/css/base.scss [hunk 1]
```diff
@@ -181,6 +181,38 @@ input[type="radio"]:not(old):checked + label > .outer > .inner {
   border-radius: 3px;
   background: white;
 }
+
+input[type="checkbox"] {
```
Comment (hatched) [inline jujugui/static/gui/src/app/assets/css/base.scss]: Do we want a style which is so generic?

**Comment by hatched** [comment]
:+1: QA :ok:

File: jujugui/static/gui/src/app/assets/css/base.scss (hunk 1 above)
Comment (huwshimi) [inline jujugui/static/gui/src/app/assets/css/base.scss]: I'm not sure if we would have a case where we don't want the checkboxes to appear the same...

File: jujugui/static/gui/src/app/assets/css/base.scss [hunk 2]
```diff
@@ -181,6 +181,38 @@ input[type="radio"]:not(old):checked + label > .outer > .inner {
...
+  text-align: center;
+  overflow: hidden;
+  -webkit-appearance: button;
+
+  &:after {
+    content: '';
+    display: none;
+    width: $size;
+    height: $size;
+    background: transparent url(juju-ui/assets/images/non-sprites/tick.svg) no-repeat center center;
```
Comment (anthonydillon) [inline jujugui/static/gui/src/app/assets/css/base.scss]: It tick looks a little low in the box, could you update the background-position to `center 2px` please.

File: jujugui/static/gui/src/app/assets/images/non-sprites/tick.svg [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+           x="31.999992"
+           y="228.3622" />
+        <path
+           style="font-style:normal;font-variant:normal;font-weight:normal;font-stretch:normal;font-size:15px;line-height:125%;font-family:Ubuntu;-inkscape-font-specification:Ubuntu;text-align:center;letter-spacing:0px;word-spacing:0px;writing-mode:lr-tb;text-anchor:middle;display:inline;fill:#ffffff;fill-opacity:1;stroke:none"
+           d="m 54.157606,231.70402 -0.173484,0.15201 -12.466718,10.91973 -7.308154,-6.26905 -2.20926,2.49843 9.519224,10.01524 14.480776,-15.23796 -1.842384,-2.07821 z"
+           id="path6399" />
+      </g>
+    </g>
+  </g>
+</svg>
```
Comment (anthonydillon) [inline jujugui/static/gui/src/app/assets/images/non-sprites/tick.svg]: Does this SVG need to be optimised? Seems like a lot of redundant info in here. We could follow up with a full optimisation of the sites SVG's if that's easier.

File: jujugui/static/gui/src/app/assets/css/base.scss (hunk 1 above)
Comment (anthonydillon) [inline jujugui/static/gui/src/app/assets/css/base.scss]: I think this is fine in this case as its the generic app-wide style

**Comment by anthonydillon** [comment]
Looks good @huwshimi, one very small positioning tweak in code. QA'ed in IE, Edge, FF, Chrome and Opera. :+1:

File: jujugui/static/gui/src/app/assets/images/non-sprites/tick.svg (hunk 3 above)
Comment (hatched) [inline jujugui/static/gui/src/app/assets/images/non-sprites/tick.svg]: @anthonydillon could you add an optimized svg to this PR for @huwshimi to use?

Comment (anthonydillon) [inline jujugui/static/gui/src/app/assets/images/non-sprites/tick.svg]: Looks like SVGO borked the optimisation. Let's go with this over and I'll try and get a less complicated one from design.

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/3169/
Test PASSed.

File: jujugui/static/gui/src/app/assets/images/non-sprites/tick.svg (hunk 3 above)
Comment (huwshimi) [inline jujugui/static/gui/src/app/assets/images/non-sprites/tick.svg]: It's OK, I managed to optimise it (down to nearly a 5th of the size).

**Comment by huwshimi** [comment]
Thanks @anthonydillon and @hatched :shipit:

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci.jujugui.org:8080/job/juju-gui-merge

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/3170/
Test PASSed.

**Pull request merged**